solr-search/src/main/java/edu/unc/lib/dl/search/solr/model/SearchState.java

```diff
@@ -135,11 +132,70 @@ public void setSearchFields(Map<String, String> searchFields) {
         this.searchFields = searchFields;
     }
 
-    public Map<String, Object> getFacets() {
+    /**
+     * Retrieve the facets for this search state. The addFacet and setFacet methods should be preferred for
+     * updating facet values.
+     * @return
+     */
+    public Map<String, List<SearchFacet>> getFacets() {
         return facets;
     }
 
-    public void setFacets(Map<String, Object> facets) {
+    /**
+     * Add a facet value to this search state
+     * @param key key of the facet to add
+     * @param value String value of the facet
+     */
+    public void addFacet(SearchFieldKeys key, String value) {
+        addFacet(new GenericFacet(key.name(), value));
+    }
+
+    /**
+     * Add a facet value to this search state
+     * @param value facet value
+     */
+    public void addFacet(SearchFacet value) {
+        List<SearchFacet> existing = facets.get(value.getFieldName());
+        if (existing == null) {
+            existing = new ArrayList<SearchFacet>();
+        }
+        facets.put(value.getFieldName(), existing);
+        existing.add(value);
+    }
+
+    /**
+     * Set the value of the specified facet to the provided value
+     * @param value value of the facet to set.
+     */
+    public void setFacet(SearchFacet value) {
+        setFacet(SearchFieldKeys.valueOf(value.getFieldName()), value);
+    }
+
+    /**
+     * Set the value of the specified facet to the provided value
+     * @param key key of the facet to add
+     * @param value value of the facet to set.
+     */
+    @SuppressWarnings("unchecked")
```

Comment:
Why is this unchecked?

Reply:
Its because of `vals = (List<SearchFacet>) value;`. I could probably split this method into two methods, one for `List<SearchFacet>` value and another for everything else